### PR TITLE
unittests: Add cpp_ctors test for C++ constructors

### PR DESCRIPTION
**Pull request by gebart**
~~Small test program to assist in debugging missing constructor calls.~~

Add a unit test for checking that global and static constructors are being called during the system start up.

### CODE REVIEW

**Comment by gebart** [comment]
This should rather be a unit test than standalone, so I'll mark it as WIP for now.

**Comment by josephnoir** [comment]
Can you write RIOT unit test for C++? Didn't look at the framework for a while, but I remember something being called from C ...

**Comment by gebart** [comment]
I guess we need an extern C trampoline function to enter the C++ domain
from the embunit main. I don't know if the assert macros work properly in
C++ though, but they look like they should work.
Den 27 jun 2016 14:19 skrev "Joseph Noir" <notifications@github.com>:

**Comment by OlegHahm** [comment]
@gebart, @josephnoir, any news here?

**Comment by gebart** [comment]
not yet

**Comment by gebart** [comment]
Refactored into a unittest. Rebased on top of #6854.

The global constructor unit test fails on native right now..

**Comment by gebart** [comment]
waiting for #6858, #6854

**Comment by gebart** [comment]
rebased on latest #6858 
@astralien3000 can you do a test run of tests/unittests for native on macos?

**Comment by astralien3000** [comment]
Done. All is fine.

File: tests/unittests/tests-cpp_ctors/tests-cpp_ctors-class.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+MyClass::MyClass()
+{
+//     std::cerr << "MyClass::MyClass()" << std::endl;
+    var = tests_cpp_ctors_magic1;
+}
+
+MyClass::MyClass(long value) : var(value)
+{
+//     std::cerr << "MyClass::MyClass(" << value << ")" << std::endl;
```
Comment (kYc0o) [inline tests/unittests/tests-cpp_ctors/tests-cpp_ctors-class.cpp]: I think the comments can be removed?

Comment (gebart) [inline tests/unittests/tests-cpp_ctors/tests-cpp_ctors-class.cpp]: oops, I'll fix

**Comment by gebart** [comment]
@kYc0o rebased, dependencies merged.

**Pull request merged**